### PR TITLE
SDK generation for azurestackhci 2022-05 release

### PR DESCRIPTION
**Pull request by lsweeney24**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/azurestackhci/Azure.ResourceManager.Hci/api/Azure.ResourceManager.Hci.netstandard2.0.cs [hunk 1]
```diff
@@ -68,7 +68,12 @@ public partial class ArcSettingData : Azure.ResourceManager.Models.ResourceData
     {
         public ArcSettingData() { }
         public Azure.ResourceManager.Hci.Models.ArcSettingAggregateState? AggregateState { get { throw null; } }
-        public string ArcInstanceResourceGroup { get { throw null; } }
+        public string ArcApplicationClientId { get { throw null; } set { } }
+        public string ArcApplicationObjectId { get { throw null; } set { } }
+        public string ArcApplicationTenantId { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/azurestackhci/Azure.ResourceManager.Hci/api/Azure.ResourceManager.Hci.netstandard2.0.cs:73]: If we know that the format of these Ids are guid, can you use directive to set them in `autorest.md`. There is existing example:
```
  - from: clusters.json
    where: $.definitions.ClusterProperties.properties
    transform: >
      $.aadClientId.format = "uuid";
      $.aadTenantId.format = "uuid";
      $.cloudId.format = "uuid";
```

**Comment by fengzhou-msft** [comment]
Can you resolve the conflicts by merging the latest main branch (ignoring the conflicts during merging) and regenerate the SDK?

File: sdk/azurestackhci/Azure.ResourceManager.Hci/api/Azure.ResourceManager.Hci.netstandard2.0.cs (hunk 1 above)
Comment (lsweeney24) [reply sdk/azurestackhci/Azure.ResourceManager.Hci/api/Azure.ResourceManager.Hci.netstandard2.0.cs:73]: Fixed

File: sdk/azurestackhci/Azure.ResourceManager.Hci/api/Azure.ResourceManager.Hci.netstandard2.0.cs [hunk 2]
```diff
@@ -113,7 +122,9 @@ public partial class HciClusterCollection : Azure.ResourceManager.ArmCollection,
     public partial class HciClusterData : Azure.ResourceManager.Models.TrackedResourceData
     {
         public HciClusterData(Azure.Core.AzureLocation location) : base (default(Azure.Core.AzureLocation)) { }
+        public string AadApplicationObjectId { get { throw null; } set { } }
         public System.Guid? AadClientId { get { throw null; } set { } }
+        public string AadServicePrincipalObjectId { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (lsweeney24) [reply sdk/azurestackhci/Azure.ResourceManager.Hci/api/Azure.ResourceManager.Hci.netstandard2.0.cs:127]: Fixed

**Pull request closed without merging**